src/app/services/app-state.service.ts

```diff
@@ -76,44 +76,63 @@ export class AppStateService {
   }
 
 
-  setRecordSelected(cinchyId: number | null, doNotReloadForm: boolean = false): void {
+  setRecordSelected(rowId: number | null, doNotReloadForm: boolean = false): void {
+    this.onRecordSelected$.next({ cinchyId: rowId, doNotReloadForm });
 
-    this._rowId = cinchyId;
-
-    this.onRecordSelected$.next({ cinchyId, doNotReloadForm });
-
-    // Update URL with the new ID, if present
-    if (window.location.search?.includes("rowId")) {
-      const queryParams = window.location.search?.substr(1).split("&").map((paramString: string) => {
+    if (rowId === null) {
+      this.deleteRowIdInQueryParams();
+    }
+    else {
+      this.updateRowIdInQueryParams(rowId);
+    }
+  }
 
-        const [key, value] = paramString.split("=");
+  updateRowIdInQueryParams(rowId: number) {
+    const messageJSON = {
+      updateCinchyURLParams:
+      {
+        rowId: rowId
+      }
+    };
 
-        if (key === "rowId") {
-          return `${key}=${cinchyId ? cinchyId.toString() : "null"}`;
-        }
-        else {
-          return paramString;
-        }
-      });
+    const message = JSON.stringify(messageJSON);
+    window.parent.postMessage(message, '*');
 
-      if (queryParams?.length) {
-        const baseUrl = window.location.href.substr(0, window.location.href.indexOf("?"));
+    const rowIdQueryParams = messageJSON['updateCinchyURLParams'];
+    const rowIdQueryString = Object.keys(rowIdQueryParams)
+      .map(key => `${encodeURIComponent(key)}=${encodeURIComponent(rowIdQueryParams[key])}`)
+      .join('&');
 
-        window.history.replaceState(window.history.state, document.title, `${baseUrl}?${queryParams.join("&")}`);
+    const queryParams = window.location.search?.substr(1).split("&").map((paramString: string) => {
+      const [key, value] = paramString.split("=");
+      if (key != "rowId") {
+        return `${key}=${value}`;
       }
+    }).join('');
+
+    if (queryParams?.length) {
+      const baseUrl = window.location.href.substr(0, window.location.href.indexOf("?"));
+      window.history.replaceState(window.history.state, document.title, `${baseUrl}?${queryParams}&${rowIdQueryString}`);
     }
-    else if (window.parent.location.search?.includes("rowId")) {
-      const messageJSON = {
-        updateCinchyURLParams:
-        {
-          rowId: cinchyId
-        }
-      };
-
-      window.parent.postMessage(JSON.stringify(messageJSON), '*');
-    }
+  }
 
-    sessionStorage.setItem("rowId", this._rowId ? this._rowId.toString() : "");
-    sessionStorage.setItem("formId", this.formId ?? "");
+  deleteRowIdInQueryParams() {
+    const messageJSON = {
+      deleteCinchyURLParams:
+        [
+          "rowId"
+        ]
+    };
+    const message = JSON.stringify(messageJSON);
+    window.parent.postMessage(message, '*');
+
+    const queryParams = window.location.search?.substr(1).split("&").map((paramString: string) => {
```

Comment:
Can you add a comment here explaining that substr is being used to remove the `?` from the querystring? It's correct to use it here, but it's not obvious.